extensions/kubernetes-client/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesConfigSourceConfig.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.kubernetes.client.runtime;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "kubernetes-config", phase = ConfigPhase.BOOTSTRAP)
+public class KubernetesConfigSourceConfig {
+
+    /**
+     * If set to true, the application will attempt to look up the configuration from the API server
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean enabled;
+
+    /**
+     * If set to true, the application will not stand up if it cannot obtain configuration from the Config Server
```

Comment:
Please avoid the "stand up" idiom :-) What's wrong with "start"?

Also, "Config Server" -> "API server"?